datalad/tests/fixtures.py

```diff
@@ -0,0 +1,155 @@
+"""utilities to assist with setting up tests"""
+
+from datalad import cfg
+from datalad.core.distributed.clone import (
+    Clone,
+    decode_source_spec
+)
+from datalad.distribution.dataset import Dataset
+from datalad.utils import (
+    better_wraps,
+    ensure_list,
+    optional_args,
+)
+from datalad.tests.utils import with_tempfile
+
+
+DATALAD_TESTS_CACHE = cfg.obtain("datalad.tests.cache")
+
+
+def get_cached_dataset(url, dataset_name=None, version=None, paths=None):
```

Comment:
BTW, I was thinking about this - where to give etc. Since I expect external tools also make use of it, and cache location to be the same, default location and path would likely not avoid conflicts (different urls into the same location). Since we don't know dataset uuId from url either - the best would be to use the same approach as git annex addurl does and create target folder name from entire url. For a test it should matter, and would likely to avoid confusion etc. Would look quite alright for all /// datasets I guess

Reply:
Not sure what you mean by "external tools". However, default clone names might be insufficient - I agree. addurl approach sounds good. But then I will eliminate `dataset_name` altogether and ensure the name does always reflect the URL.

Reply:
Re external: so we could use it eg in heudiconv by importing from datalad . I don't see an immediate need for name to stay, so removal sounds good to me

Reply:
Done. Simply replaced `/,:,?` by `_` for now. Thought about hashing the URL instead, but this would become more complicated if we add version to the URL itself. Guess we need to see how this evolves. Will need more sophisticated URL handling as we make use of those tools for `ria+` URLs anyway.